lib/events/dynamoevents/dynamoevents.go

```diff
@@ -295,8 +300,9 @@ func New(ctx context.Context, cfg Config, backend backend.Backend) (*Log, error)
 		return nil, trace.Wrap(err)
 	}
 
-	// Migrate the table according to RFD 24 if it still has the old schema.
-	go b.migrateRFD24WithRetry(ctx)
+	// Migrate the table.
+	go b.migrateWithRetry(ctx, b.migrateRFD24, "migrateRFD24")
+	go b.migrateWithRetry(ctx, b.migrateFieldsMap, "migrateFieldsMap")
```

Comment:
I'm not totally sure about this but at first glance if we run two migration simultaneously and both `migrateRFD24` and `migrateFieldsMap` needs to be preformed there is a corner case when both ongoing migration scans the same item and tries to update it. So in that case there will be a write race condition thus an item will either contain updated `CreatedAt` or `FieldsMap` fields because update is done by `upsert` operation.

Reply:
Due to the [component compatibility policy](https://goteleport.com/docs/setup/operations/upgrading/#component-compatibility) I don't think this scenario can really occur in the wild -- `migrateRFD24` could be removed in 8.0 IIUC, I just didn't want to mess with it. To be absolutely sure, I'll update this so that the migration procedures are run in sequence.